### PR TITLE
Add 'wpsc_duplicate_product_thumbnail' filter.

### PR DESCRIPTION
**Pull request by benhuson**
For issue #1672

When duplicating a product, this is the workflow:

 - Attached images are duplicated to the new product and featured image set.
 - If images are set not to copy via the 'wpsc_duplicate_product_attachment' the featured image is not copied either.

The 'wpsc_duplicate_product_thumbnail' allows you to alter the above outcome.

### CODE REVIEW

**Comment by benhuson** [comment]
Here's an example:

```<?php

// Do not copy images when a product is duplicated
function test_wpsc_duplicate_product_attachment( $thumbnail, $attachment, $new_post_id ) {
	return false;
}
add_filter( 'wpsc_duplicate_product_attachment', 'test_wpsc_duplicate_product_attachment', 10, 3 );

// Set the featured image to the same as the original product
function test_wpsc_duplicate_product_thumbnail( $thumbnail_id, $original_thumbnail_id, $post_id, $new_post_id ) {
	return $original_thumbnail_id;
}
add_filter( 'wpsc_duplicate_product_thumbnail', 'test_wpsc_duplicate_product_thumbnail', 10, 4 );

?>
```

File: wpsc-admin/admin.php [hunk 1]
```diff
@@ -1474,6 +1477,46 @@ function wpsc_duplicate_product_meta( $id, $new_id ) {
...
+ *
+ * If the featured image is not attached to the product the featured image ID
+ * remains the same as the original product.
+ *
+ * This function will remove the featured image if the image is not attached to
+ * the duplicated product and offers the opportunity to change the featured image
+ * of the duplicated product via the 'wpsc_duplicate_product_thumbnail' filter.
+ *
+ * @param   integer  $id      Product ID.
+ * @param   integer  $new_id  Duplicated product ID.
```
Comment (JustinSainton) [inline wpsc-admin/admin.php]: These need to be updated to reflect the updated parameter names (`$post_id` and `$new_post_id`)

**Comment by JustinSainton** [comment]
Everything looks pretty good, @benhuson!  Just a minor docs issue then we can merge.

**Comment by JustinSainton** [comment]
And FWIW, in the "do not copy images" example, one could do `add_filter( 'wpsc_duplicate_product_attachment', '__return_false' )`

**Comment by benhuson** [comment]
Yep, I always forget you can `__return_false `

**Pull request merged**